src/models/cost_functions/MarketBidCost.jl

```diff
@@ -0,0 +1,106 @@
+"""
+    mutable struct MarketBidCost <: OperationalCost
+        no_load_cost::Float64
+        start_up::NamedTuple{(:hot, :warm, :cold), NTuple{3, Float64}}
+        shut_down::Float64
+        incremental_offer_curves::Union{Nothing, IS.TimeSeriesKey, PiecewiseLinearData}  # TODO update docs given struct
+        decremental_offer_curves::Union{Nothing, IS.TimeSeriesKey, PiecewiseLinearData}
+        ancillary_service_offers::Vector{Service}
+    end
+
+Data Structure Operational Cost to reflect market bids of energy and ancilliary services for any asset.
+Compatible with most US Market bidding mechanisms that support demand and generation side.
+
+# Arguments
+- `no_load_cost::Float64`: no load cost
+- `start_up::NamedTuple{(:hot, :warm, :cold), NTuple{3, Float64}}`: start-up cost at different stages of the thermal cycle. Warm is also refered as intermediate in some markets
+- `shut_down::Float64`: shut-down cost, validation range: `(0, nothing)`, action if invalid: `warn`
+- `incremental_offer_curves::Union{Nothing, IS.TimeSeriesKey, PiecewiseLinearData}`: Sell Offer Curves data, can be a time series or a fixed PiecewiseLinearData
+- `decremental_offer_curves::Union{Nothing, IS.TimeSeriesKey, PiecewiseLinearData}`: Buy Offer Curves data, can be a time series or a fixed PiecewiseLinearData
+- `ancillary_service_offers::Vector{Service}`: Bids for the ancillary services
+"""
+@kwdef mutable struct MarketBidCost <: OperationalCost
+    no_load_cost::Float64
+    """start-up cost at different stages of the thermal cycle.
+    Warm is also referred to as intermediate in some markets"""
+    start_up::StartUpStages
+    "shut-down cost"
+    shut_down::Float64
```

Comment:
Based on the latest analysis of the curves, these fields should be `Union{Float64, IS.TimeSeriesKey}` to support being read from time series. This we can implement in a later PR after merging this one and time series

Reply:
Added to https://github.com/NREL-Sienna/PowerSystems.jl/pull/1071#discussion_r1579880930